extensions/artifacts/org.wso2.developerstudio.codenvy.ext.appfactory/src/main/java/org/wso2/developerstudio/codenvy/ext/appfactory/server/authentication/Authenticator.java

```diff
@@ -0,0 +1,64 @@
+/*
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.wso2.developerstudio.codenvy.ext.appfactory.server.authentication;
+
+import org.wso2.developerstudio.codenvy.ext.appfactory.server.client.HTTPAppFactoryClient;
+import org.wso2.developerstudio.codenvy.ext.appfactory.server.exception.AppFactoryServerException;
+import org.wso2.developerstudio.codenvy.ext.appfactory.shared.dto.AppFactoryLoginInfo;
+import org.wso2.developerstudio.codenvy.ext.appfactory.shared.dto.AppFactoryLoginResponse;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class handles all the authentication related stuffs from WSO2 Developer Studio to WSO2 AppFactory/ AppCloud
+ */
+public class Authenticator {
```

Comment:
Hi Prabath,
Thanks for the feedback.
Before I explain the reasons behind making this singleton, I would like to point you to the latest code of Authenticator.java class at https://github.com/rajeevanWSO2/cloud-dev-studio/blob/master/extensions/artifacts/org.wso2.developerstudio.codenvy.ext.appfactory/src/main/java/org/wso2/developerstudio/codenvy/ext/appfactory/server/authentication/Authenticator.java
The class you were looking is the early stage of the App Factory extension development and I committed changes partially as the development happens .
Basically we wanted to have a single instance of Authenticator.java class per an instance of DevStudio, because only a single user of App Factory can login to the App Factory from DevStudio at a time (considering standalone version of DevStudio) and also if the cookie expires after some time (it’s some less time and configured in App Cloud/ App Factory) after a user logged in, we can't ask the user to enter the login credential for each user operation like check in projects, view application details etc. It is the duty of DevStudio to remember the credentials and do a re login automatically and connect to App Factory. This is the use case scenario. 
So we wanted to have an instance so that it can keep all the login data at the first login and use those data if a re login operation needed. So I created a singleton class to make sure only a single instance will be created in a DevStudio instance and that instance will be used in the whole process. Please note that re login method will not have any arguments as parameters and will use the instance variables of Authenticator class assigned from 'authenticate' method. Also, we have not yet implemented the re login method in Authenticator.java class and will be implemented in near future. I will add a comment top of the class explaining this design pattern. Please correct me if I am wrong.